Sources/ClientRuntime/Networking/Http/URLSession/URLSessionHTTPClient.swift

```diff
@@ -485,7 +497,12 @@ public final class URLSessionHTTPClient: HTTPClient {
         guard let url = components.url else { throw URLSessionHTTPClientError.incompleteHTTPRequest }
         var urlRequest = URLRequest(url: url, timeoutInterval: self.connectionTimeout)
         urlRequest.httpMethod = request.method.rawValue
-        urlRequest.httpBodyStream = httpBodyStream
+        switch body {
+        case .stream(let bridge):
+            urlRequest.httpBodyStream = bridge.inputStream
+        case .data(let data):
+            urlRequest.httpBody = data
+        }
```

Comment:
This is the key: the urlRequest only uses a stream when the Data isn't already in memory.